lib/rubocop/cop/variable_force.rb

```diff
@@ -354,6 +353,16 @@ def descendant_reference(node)
         end
       end
 
+      def reference_assignments(loop_assignments, node)
+        # If inside a case statement, mark all as referenced.
+        # Otherwise, mark only the last assignment as referenced.
+        if loop_assignments.first.node.each_ancestor(:case).any?
```

Comment:
Is it necessary to account for `case-match`?

```suggestion
        if loop_assignments.first.node.each_ancestor(:case, :case_match).any?
```

Reply:
You're right, it's necessary as well. I've added it. Thank you.